experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -766,25 +774,46 @@ export type TypedFieldInner<
  */
 export type TypedNodeUnion<TTypes extends AllowedTypes> =
 	TTypes extends InternalTypedSchemaTypes.FlexList<TreeNodeSchema>
-		? InternalTypedSchemaTypes.ArrayToUnion<
-				TypeArrayToTypedTreeArray<
-					Assume<
-						InternalTypedSchemaTypes.ConstantFlexListToNonLazyArray<TTypes>,
-						readonly TreeNodeSchema[]
-					>
-				>
-		  >
+		? TypedNodeUnionHelper<TTypes>
 		: TreeNode;
 
+/**
+ * Helper for implementing TypedNodeUnionHelper.
```

Comment:
```suggestion
 * Helper for implementing TypedNodeUnion.
```

Reply:
Or, even better
```suggestion
 * Helper for implementing {@link TypedNodeUnion}.
```

Reply:
That link won't work as its an internal type (exported, but in our internalTypes scope). Well, it will work in intelisense but API-Extractor will object. IIRC such things are impossible to link.